docs/designers-developers/developers/themes/theme-json.md

```diff
@@ -0,0 +1,219 @@
+# Themes & Block Editor: Experimental theme.json
+
+This is intended to document the current direction and work in progress about how themes can hook into the various sub-systems that the Block Editor provides.
+
+* Rationale
+* Specification
+* Current Status
+
+## Rationale
+
+The Block Editor surface API has evolved at different velocities, and it's now at a point where is showing some growing pains, specially in areas that affect themes. Examples of this are: the ability to [control the editor programmatically](https://make.wordpress.org/core/2020/01/23/controlling-the-block-editor/), or [a block style system](https://github.com/WordPress/gutenberg/issues/9534) that facilitates user, theme, and core style preferences.
+
+This describes the current efforts to consolidate the various APIs into a single point: a `experimental-theme.json` file.
+
+When this file is present and the Full Site Editing experiment is enabled, a few Block Editor mechanisms that are activated.
+
+### Presets become CSS Custom Properties
+
+Presets such as [color palettes](https://developer.wordpress.org/block-editor/developers/themes/theme-support/#block-color-palettes), [font sizes](https://developer.wordpress.org/block-editor/developers/themes/theme-support/#block-font-sizes), and [gradients](https://developer.wordpress.org/block-editor/developers/themes/theme-support/#block-gradient-presets) will be enqueued as CSS Custom Properties for themes to use.
+
+These are enqueued for both in the front-end and the site editor.
+
+### Some block styles are managed
+
+By providing the block style properties in a structured way, the Block Editor can "manage" the CSS that comes from different origins (user, theme, and core CSS), reducing the amount of CSS loaded in the page and preventing specificity wars due to the competing needs of the components involved (themes, blocks, plugins).
+
+### Individual features can be controlled per block
+
+The Block Editor already allows the control of specific features such as alignment, drop cap, whether it's present in the inserter, etc. The goal is that these are also available for themes to control.
+
+## Specification
+
+The specification for the `experimental-theme.json` follows the three main functions described in the section above: presets, styles, and features.
+
+```
+{
+  presets: {
+    color: [ ... ],
+    font-size: [ ... ],
+    gradient: [ ... ],
+  },
+  styles: { ... },
+  features: { ... }
```

Comment:
I  still have a small  concern about the "presets"  and "features".

For  instance to disable colors entirely you'd have to do:

```
presets : {
   color: []
},
features: {
   customColors: false,
}
```
So two separate locations to  "control" the editor.

Reply:
Yeah, I agree, that'd be not ideal. This is not implemented yet but the way I imagine this could work is by compressing it to:

```
presets : {
   color: false
}
```

Reply:
it's still two top different top level keys to disable a feature though.